### PR TITLE
Feature/#14 settings

### LINKED ISSUE #14: Settings
Create settings for the application. Make them changeable by changing the settings file or through the application.

- [x] Create and save settings.xml
- [x] Save data:
  - [x] Transfer folder path
  - [x] Server IP and Port
- [x] Be able to have multiple setting configurations
- [x] Choose configuration on startup, be able to skip choice by selecting a default
- [x] Load configuration after startup
- [x] Be able to reset default or select a new one
- [x] Test all settings

### PR DESCRIPTION
**Pull request by TheoCremers**
closes #14

### CODE REVIEW

File: DNDinventory/View/MainWindow.xaml.cs [hunk 1]
```diff
@@ -31,5 +31,9 @@ public MainWindow()
 
             Closing += _viewModel.OnWindowClosing;
         }
+
+        private void MenuItem_Click (object sender, RoutedEventArgs e) {
```
Comment (vman26) [inline DNDinventory/View/MainWindow.xaml.cs]: Remove as not used

File: DNDinventory/View/MainWindow.xaml [hunk 2]
```diff
@@ -34,8 +34,15 @@
                 </StackPanel>
                 <MenuItem Header="{Binding ConnectText}" Command="{Binding ConnectCommand}"/>
             </MenuItem>
-            <MenuItem Header="Settings">
+            <MenuItem Header="Settings" Click="MenuItem_Click">
```
Comment (vman26) [inline DNDinventory/View/MainWindow.xaml]: Remove as not used

File: DNDinventory/View/MainWindow.xaml [hunk 3]
```diff
@@ -34,8 +34,15 @@
...
                 <MenuItem Header="{Binding SetFolderOutputTxt}" Command="{Binding SetOutputFolderCommand}"/>
+                <Separator HorizontalAlignment="Left" Height="1" Width="138"/>
```
Comment (vman26) [inline DNDinventory/View/MainWindow.xaml]: Why not using default separator?

File: DNDinventory/View/MainWindow.xaml [hunk 4]
```diff
@@ -34,8 +34,15 @@
...
+                <MenuItem Header="Save" Command="{Binding SaveSettingsCommand}"/>
+                <MenuItem Header="Load" Command="{Binding LoadSettingsCommand}"/>
+                <Separator HorizontalAlignment="Left" Height="1" Width="138"/>
```
Comment (vman26) [inline DNDinventory/View/MainWindow.xaml]: See comment above

File: DNDinventory/ViewModel/MainViewModel.cs [hunk 5]
```diff
@@ -121,6 +123,20 @@ public string SetFolderOutputTxt
...
+        private string settingsFileLocationTxt;
+        public string SettingsFileLocationTxt {
+            get {
+                return settingsFileLocationTxt;
+            }
+            set {
+                if (settingsFileLocationTxt != value) {
+                    settingsFileLocationTxt = $"Current settings file ({value})";
+                    OnPropertyChange("SettingsFileLocationTxt");
+                    Properties.Settings.Default.SettingsFileLocation = value;
```
Comment (vman26) [inline DNDinventory/ViewModel/MainViewModel.cs:135]: Latest used, not default by design ;-)

File: DNDinventory/SettingsFileHandler.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        public string port = "100";
+        public string outputFolder = "Transfers";
+    }
+
+    public class SettingsFileHandler {
+        public SavedSettings currentSettings = new SavedSettings();
+        XmlSerializer ser = new XmlSerializer(typeof(SavedSettings));
+        
+        public void WriteToXml (string filename) {
+            TextWriter writer = new StreamWriter(filename);
```
Comment (vman26) [inline DNDinventory/SettingsFileHandler.cs]: Use using(){} of C# this way when an error occurs the writer will be closed. The current implementation will force exit the function.

File: DNDinventory/SettingsFileHandler.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            ser.Serialize(writer, currentSettings);
+            writer.Close();
+        }
+
+        public void ReadFromXml (string filename) {
+            if (File.Exists(filename)) {
+                FileStream fs = new FileStream(filename, FileMode.Open);
```
Comment (vman26) [inline DNDinventory/SettingsFileHandler.cs]: Same here

File: DNDinventory/SettingsFileHandler.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+namespace DNDinventory {
+    public class SavedSettings {
+        // Set defaults here
+        public string host = "localhost";
+        public string port = "100";
+        public string outputFolder = "Transfers";
+    }
+
+    public class SettingsFileHandler {
+        public SavedSettings currentSettings = new SavedSettings();
```
Comment (vman26) [inline DNDinventory/SettingsFileHandler.cs]: Create constructor for creating the defaults.

```
void SettingsFileHandler()
{
    currentSettings = new SavedSettings();
    etc.
}
```

**Review by vman26 — CHANGES_REQUESTED** [review]
Small changes to make. Yet functional feature.

File: DNDinventory/ViewModel/MainViewModel.cs (hunk 5 above)
Comment (TheoCremers) [reply DNDinventory/ViewModel/MainViewModel.cs:135]: intentional design, let the users test it first.

Comment (vman26) [reply DNDinventory/ViewModel/MainViewModel.cs:135]: Default is latest opened file. Agreed and see reactions

**Review by vman26 — APPROVED** [review]

**Pull request merged**